src/main/java/org/testng/TestRunner.java

```diff
@@ -862,6 +866,13 @@ private void privateRun(XmlTest xmlTest) {
     return result;
   }
 
+  private static Map<String, String> getParameters(IMethodInstance im) {
+    XmlTest xmlTest = im.getMethod().getXmlTest();
+    Map<String, String> params = xmlTest.getAllParameters();
```

Comment:
findMethodParameters is already doing it: https://github.com/cbeust/testng/blob/8d1c11bc267ea52157e20ebf6781d1ca9fe7ed1c/src/main/java/org/testng/internal/BaseTestMethod.java#L780